src/python/dxpy/nextflow/nextflow_builder.py

```diff
@@ -0,0 +1,151 @@
+    :type resources_dir: str or Path
+    :param profile: Custom NF profile, for more information visit https://www.nextflow.io/docs/latest/config.html#config-profiles
+    :type profile: string
+
+    Creates files necessary for creating an applet on the Platform, such as dxapp.json and a source file. These files are created in '.dx.nextflow' directory.
+    """
+    assert os.path.exists(resources_dir)
+    inputs = []
+    os.makedirs(".dx.nextflow", exist_ok=True)
+    dxapp_dir = os.path.join(resources_dir, '.dx.nextflow')
+    if os.path.exists(f"{resources_dir}/nextflow_schema.json"):
+        inputs = prepare_inputs(f"{resources_dir}/nextflow_schema.json")
+    dxapp_content = get_nextflow_dxapp(inputs)
+    exec_content = get_nextflow_src(inputs=inputs, profile=profile)
+    copy_tree(get_template_dir(), dxapp_dir)
+    write_dxapp(dxapp_dir, dxapp_content)
+    write_exec(dxapp_dir, exec_content)
+    return dxapp_dir
+
+
+def prepare_inputs(schema_file):
+    """
+    :param schema_file: path to nextflow_schema.json file
+    :type schema_file: str or Path
+    :returns: DNAnexus datatype used in dxapp.json inputSpec field
+    :rtype: string
+    Creates DNAnexus inputs (inputSpec) from Nextflow inputs.
+    """
+    def get_default_input_value(key):
+        input_items = {
+            "hidden": False,
+        }
+        if key in input_items:
+            return input_items[key]
+        raise Exception("Default value for key {} is not given.".format(key))
+
+    def get_dx_type(nf_type):
+        types = {
+            "string": "str",
+            "integer": "int",
+            "number": "float",
+            "boolean": "boolean",
+            "object": "hash"
+            # TODO: add directory + file + path
+        }
+        if nf_type in types:
+            return types[nf_type]
+        # TODO: raise Exception after file+directory is implemented
+        return "string"
+        # raise Exception(f"type {nf_type} is not supported by DNAnexus")
+
+    inputs = []
+    with open(schema_file, "r") as fh:
+        schema = json.load(fh)
+    for d_key, d_schema in schema.get("definitions", {}).items():
+        required_inputs = d_schema.get("required", [])
+        for property_key, property in d_schema.get("properties", {}).items():
+            dx_input = {}
+            dx_input["name"] = property_key
+            dx_input["title"] = dx_input['name']
+            if "help_text" in property:
+                dx_input["help"] = property.get('help_text')
+            if "default" in property:
+                dx_input["default"] = property.get("default")
+            dx_input["hidden"] = property.get('hidden', get_default_input_value('hidden'))
+            dx_input["class"] = get_dx_type(property_key)
+            if property_key not in required_inputs:
+                dx_input["optional"] = True
+                dx_input["help"] = "(Optional) {}".format(dx_input["help"])
```

Comment:
```suggestion
                dx_input["help"] = "(Optional) {}".format(dx_input.get("help","")
```

Reply:
I added condition that checks whether help exists. Proposed solution would cause empty help message instead of no help message in dxapp.json